commit 5977610de1aa87630e40a299a2d90fb7cd00bf7c
Author: Christoph M. Becker <cmbecker69@gmx.de>
Date:   Mon Aug 9 12:48:21 2021 +0200

    Fix #74544: Integer overflow in mysqli_real_escape_string()
    
    The patch has been provided by @johannes.
    
    Closes GH-7353.

diff --git a/ext/mysqli/mysqli_api.c b/ext/mysqli/mysqli_api.c
index 9896ab8eda..8612a60432 100644
--- a/ext/mysqli/mysqli_api.c
+++ b/ext/mysqli/mysqli_api.c
@@ -1971,7 +1971,7 @@ PHP_FUNCTION(mysqli_real_escape_string) {
 	}
 	MYSQLI_FETCH_RESOURCE_CONN(mysql, mysql_link, MYSQLI_STATUS_VALID);
 
-	newstr = zend_string_alloc(2 * escapestr_len, 0);
+	newstr = zend_string_safe_alloc(2, escapestr_len, 0, 0);
 	ZSTR_LEN(newstr) = mysql_real_escape_string_quote(mysql->mysql, ZSTR_VAL(newstr), escapestr, escapestr_len, '\'');
 	newstr = zend_string_truncate(newstr, ZSTR_LEN(newstr), 0);
 
